osidb/tests/test_flaw.py

```diff
@@ -917,6 +917,49 @@ def test_validate_major_incident_summary(
         else:
             assert flaw1.save() is None
 
+    @pytest.mark.parametrize(
+        "state,should_raise",
+        [
+            (Flaw.FlawMajorIncident.NOVALUE, False),
+            (Flaw.FlawMajorIncident.REQUESTED, False),
+            (Flaw.FlawMajorIncident.REJECTED, False),
+            (Flaw.FlawMajorIncident.APPROVED, False),
+            (Flaw.FlawMajorIncident.CISA_APPROVED, False),
+            (Flaw.FlawMajorIncident.INVALID, True),
+        ],
+    )
+    def test_validate_major_incident_state(self, state, should_raise):
+        """
+        Tests that a flaw has a valid Major Incident state.
+        """
+        flaw = FlawFactory.build(
+            is_major_incident=True,
+            major_incident_state=state,
+            mitigation="mitigation",
+            statement="statement",
+            embargoed=False,
+        )
+        flaw.save(raise_validation_error=False)
+
+        AffectFactory(flaw=flaw)
+        FlawReferenceFactory(
+            flaw=flaw,
+            type=FlawReference.FlawReferenceType.ARTICLE,
+            url="https://access.redhat.com/link123",
+        )
+        FlawMetaFactory(
+            flaw=flaw,
+            type=FlawMeta.FlawMetaType.REQUIRES_SUMMARY,
+            meta_attr={"status": "+"},
+        )
```

Comment:
Why do we define affect, reference, and meta here? It seems unnecessary with respect to the `_validate_major_incident_state` validation. Maybe the affect is necessary to make the flaw generally valid but why the rest?

Reply:
I take this back... I realized that Major Incident has extra requirements. Ignore.